### PR TITLE
Refactor event groups

### PR DESCRIPTION
**Pull request by stevekinney**
Simplifies event groups to use primitive data structures, improves types for events.

### CODE REVIEW

File: src/lib/utilities/get-event-classification.ts [hunk 1]
```diff
@@ -89,42 +97,47 @@ const getName = (event: EventOrGroup): string => {
...
 };
 
 const getTime = (event: EventOrGroup): string => {
   let ts: Timestamp;
 
   if (isEvent(event)) ts = event.eventTime;
   if (isPendingActivity(event)) ts = event.lastStartedTime;
-  if (isEventsGroup(event)) ts = event.last.eventTime;
+  if (isEventGroup(event)) {
+    console.log(event, getLastEvent(event));
```
Comment (GiantRobots) [inline src/lib/utilities/get-event-classification.ts:109]: console.log

Comment (stevekinney) [reply src/lib/utilities/get-event-classification.ts]: Good catch—fixed!

File: src/routes/namespaces/[namespace]/workflows/[workflow]/[run]/history/compact/activity-[eventId]/__layout.svelte [hunk 2]
```diff
@@ -58,18 +54,16 @@
...
+      {#each events as event}
+        <li>
+          <a
+            href={getHref(group, event)}
+            class:active={$page.path === getHref(group, event)}
+          >
+            {event.eventType}
+          </a>
+        </li>
+      {/each}
```
Comment (GiantRobots) [inline src/routes/namespaces/[namespace]/workflows/[workflow]/[run]/history/compact/activity-[eventId]/__layout.svelte:66]: Might want to poke Candace about an empty state here? I don't think our model _should_ have workflows without events but stranger things have happened.

File: src/routes/namespaces/[namespace]/workflows/[workflow]/[run]/history/compact/__layout.svelte [hunk 3]
```diff
@@ -2,26 +2,24 @@
   import type { LoadInput } from '@sveltejs/kit';
 
   export const load = async ({ stuff }: LoadInput) => {
-    const { events } = stuff;
-
-    console.log({ events });
+    const { events } = stuff as { events: HistoryEventWithId[] };
```
Comment (GiantRobots) [inline src/routes/namespaces/[namespace]/workflows/[workflow]/[run]/history/compact/__layout.svelte:5]: Man shoving stuff into stuff seems to be potentially difficult to type. It'd be nice if we didn't have to as this and we had some kind of global stuff type like we did with getContext but not super sure how to implement this. We definitely need to parse what is in stuff as a human so it's hard to involve the compiler on this.

I wonder if there's a empty/404 case we're missing here.

File: src/routes/namespaces/[namespace]/workflows/[workflow]/[run]/history/compact/activity-[eventId]/__layout.svelte [hunk 4]
```diff
@@ -10,10 +10,16 @@
       eventId,
     } = page.params;
 
-    const { events } = stuff;
+    const { eventGroups } = stuff;
+    const group = eventGroups.find(({ id }) => id === eventId);
+
+    if (!group) return { status: 404 };
```
Comment (GiantRobots) [inline src/routes/namespaces/[namespace]/workflows/[workflow]/[run]/history/compact/activity-[eventId]/__layout.svelte:16]: 👌

**Review by GiantRobots — APPROVED** [review]

**Pull request merged**